app/js/streaming/ProtectionExtensions.js

```diff
@@ -84,41 +92,86 @@ MediaPlayer.dependencies.ProtectionExtensions.prototype = {
         if (null !== cdmData) {
             return mediaKeys.createSession(mediaCodec, initData, cdmData);
         }
+
         return mediaKeys.createSession(mediaCodec, initData);
     },
 
     getKeySystems: function (protectionData) {
         var self = this,
-            _protectionData = protectionData,
-            getLAUrl = function (laUrl, keysystem) {
-                if (protectionData[keysystem] !== undefined) {
-                    if (protectionData[keysystem].laUrl !== null && protectionData[keysystem].laUrl !== '') {
-                        return protectionData[keysystem].laUrl;
+            _stringUtils = self.stringMethods,
+            _protectionData = {
+                licenseRequest: {
+                    'com.microsoft.playready': {
+                        cdmData: null,
+                        laUrl: null,
+                        headers: null
+                    },
+                    'com.widevine.alpha': {
+                        cdmData: null,
+                        laUrl: null,
+                        headers: null
                     }
                 }
-                return laUrl;
             },
-            playreadyGetUpdate = function (event) {
-                var decodedChallenge = null,
+            getUpdate = function (event, keySystem, laUrl, headers, message) {
+                var xhr = new XMLHttpRequest();
+
+                xhr.open('POST', laUrl, true);
```

Comment:
Hey @nadimix,

It works for me, just tried it. One thing I can think of off the top of my head is that there are two slight differences in how the protectionData is initially set in my branch versus what is in the dev branch now which could cause laUrl() to give you null. The laUrl, cdmData & headers are wrapped in a licenseRequest object and the function is named setProtectionData, ex:
```
player.setProtectionData({
    licenseRequest: {
        'com.microsoft.playready': {
            cdmData: null,
            laUrl: null,
            headers: null
        },
        'com.widevine.alpha': {
            cdmData: null,
            laUrl: null,
            headers: null
        }
    }
});
```